lib/publiccloud/provider.pm

```diff
@@ -43,6 +43,27 @@ sub init {
     $self->create_ssh_key();
 }
 
+=head2 generate_basename
+
+    Call: $self->generate_basename();
+
+    openqa test product name composition, based on basic product parameters values and eventual modifiers.
+
+    Returns a string-name containing distri, version, flavor, arch.
+
+=cut
+
+sub generate_basename {
```

Comment:
This is not used outside Azure.
I suggest to just create `publiccloud::azure::generate_offer`.

Reply:
but it COULD be used: wdyt to allow future use elsewhere if needed?

Reply:
PLease @pdostal, could we try to merge it today ?

Reply:
It just adds a lot of complications for nothing :(

Reply:
moving a routine from a module to another is _complication_ ?

Reply:
moving a routine from a module to another is complication ? ok, no merge today.